src/_P240_Vindriktning.ino

```diff
@@ -0,0 +1,376 @@
+//#######################################################################################################
+//################################## Plugin 240: Dust Sensor Ikea Vindriktning ##########################
+//#######################################################################################################
+/*
+  Plugin is based upon various open sources on the internet. Plugin uses the serial lib to access a serial port
+  This plugin was written by Mark Flesch
+
+  This plugin reads the particle concentration from the PM1006 sensor in the Ikea Vindriktning
+  The original Ikea processor is controlling the PM1006 sensor, the plugin is eavesdropping the responses
+  DevicePin1 - RX on ESP, TX on PM1006
+  DevicePin2 - TX on ESP, RX on PM1006 (optional, currently not accessed by the plugin)
+*/
+
+
+// #include section
+#include "_Plugin_Helper.h"
+#ifdef USES_P240
+# include <ESPeasySerial.h>
+
+// Standard plugin defines
+#define PLUGIN_240
+#define PLUGIN_ID_240     240                   // plugin id
+#define PLUGIN_NAME_240   "Dust - Vindriktning" // "Plugin Name" is what will be dislpayed in the selection list
+#define PLUGIN_VALUENAME1_240 "PM2.5"           // variable output of the plugin. The label is in quotation marks
+#define PLUGIN_240_DEBUG  true                  // set to true for extra log info in the debug
+
+
+//   PIN/port configuration is stored in the following:
+//   CONFIG_PIN1 - Used by plugin_Helper_serial (RX pin) 
+//   CONFIG_PIN2 - Used by plugin_Helper_serial (TX pin)
+//   CONFIG_PIN3 - Not used
+//   CONFIG_PORT - Used by the plugin_Helper_serial (serialType)
+//
+//   Custom configuration is stored in the following:
+//   PCONFIG(x)
+//   x can be between 1 - 8 and can store values between -32767 - 32768 (16 bit)
+//
+//   N.B. these are aliases for a longer less readable amount of code. See _Plugin_Helper.h
+//
+//
+//   PCONFIG_LABEL(x) is a function to generate a unique label used as HTML id to be able to match 
+//                    returned values when saving a configuration.
+
+// Make accessing specific parameters more readable in the code
+// #define P240_BAUDRATE           PCONFIG_LONG(0)
+// #define P240_BAUDRATE_LABEL     PCONFIG_LABEL(0)
+
+// States for statemachine used to decode received message
+typedef enum {
+    PM1006_HEADER,
+    PM1006_LENGTH,
+    PM1006_DATA,
+    PM1006_CHECK
+} pm1006_state_t;
+
+// Global variables TODO: attach them to the task to allow multiple instances
+  ESPeasySerial *easySerial = nullptr;
+  const int bufferSize = 20;
+  char serialRxBuffer[bufferSize];
+  char debugBuffer[3*bufferSize];
+  int rxBufferPtr = 0;
+  int rxChecksum = 0;
+  int _index = 0;
+  int _rxlen = 0;
+  int pm25 = 0;
+  pm1006_state_t rxState = PM1006_HEADER;
```

Comment:
These are really "common" variable names, which is quite tricky when used as global variables.
Can you at least add `P240_` as prefix?

And this P240 nr must also be changed to match the next unused one.